sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -248,8 +248,62 @@ def test_absolute_error_sample_weight():
     gbdt.fit(X, y, sample_weight=sample_weight)
 
 
+@pytest.mark.parametrize("y", [([1.0, -2.0, 0.0]), ([0.0, 1.0, 2.0])])
+def test_gamma_y_positive(y):
+    # Test that ValueError is raised if any y_i <= 0.
+    err_msg = r"loss='gamma' requires strictly positive y."
+    gbdt = HistGradientBoostingRegressor(loss="gamma", random_state=0)
+    with pytest.raises(ValueError, match=err_msg):
+        gbdt.fit(np.zeros(shape=(len(y), 1)), y)
+
+
+def test_gamma():
+    # For Gamma distributed target, an HGBT with Gamma loss should give better results
+    # than an HGBT with squared error, measured in Gamma deviance as metric/score.
+    # Note that squared error could potentially predict negative values which is
+    # invalid (np.inf) for the Gamma deviance. A Poisson HGBT (having a log link)
+    # would not have that defect.
+    # Important note: It seems that a Poisson HGBT almost always has better
+    # out-of-sample performance than the Gamma HGBT, measured in Gamma deviance.
+    # LightGBM shows the same behaviour. The exact origin is unclear.
```

Comment:
Have you made this observation for this configuration or is it something that you have observed generally?

Also, I propose adding a remark for the choice of MSE in an extra line.
```suggestion
    # Important note: It seems that a Poisson HGBT almost always has better
    # out-of-sample performance than the Gamma HGBT, measured in Gamma deviance.
    # LightGBM shows the same behaviour. The exact origin is unclear.
    # Hence, we here use the MSE loss instead the Poisson for fairer comparison.
```

Reply:
But we also evaluate with the gamma deviance, never MSE.

Reply:
I went with ` Hence, we only compare to a squared error HGBT, but not to a Poisson deviance HGBT.`